mono/utils/mono-threads-coop.c

```diff
@@ -166,6 +166,7 @@ return_stack_ptr (gpointer *i)
 static void
 copy_stack_data (MonoThreadInfo *info, MonoStackData *stackdata_begin)
 {
+#if defined (HOST_WATCHOS) || defined (HOST_TVOS)
```

Comment:
Can we use runtime if against constants? i.e. so this code continues get type checked on regular builds?

Reply:
That's a good point.  Actually the stack copying / scanning code  should probably also run on a regular basis on a more mundane platform so that we get at least some reassurance that it's working.

I'll rework the PR.